app/views/thredded/preferences/_form.html.erb

```diff
@@ -11,29 +11,43 @@
   <ul class="thredded--form-list">
     <li>
       <%= f.label :follow_topics_on_mention do %>
-        <%= f.check_box :follow_topics_on_mention %>
+        <%= f.check_box :follow_topics_on_mention, 'data-thredded-bound-messageboard-pref' => 'user_preferences_form[messageboard_follow_topics_on_mention]' %>
         <%= t 'thredded.preferences.form.follow_topics_on_mention.label' %>
         <p class="thredded--form-list--hint">
           <%= t 'thredded.preferences.form.follow_topics_on_mention.hint' %>
         </p>
       <% end %>
     </li>
     <li>
-      <%= f.label :followed_topic_emails do %>
-        <%= f.check_box :followed_topic_emails %>
-        <%= t 'thredded.preferences.form.followed_topic_emails.label' %>
-        <p class="thredded--form-list--hint">
-          <%= t 'thredded.preferences.form.followed_topic_emails.hint' %>
-        </p>
-      <% end %>
+      <label><%= t 'thredded.preferences.form.notifications_for_followed_topics.label' %></label>
+      <!--<p class="thredded&#45;&#45;form-list&#45;&#45;hint">-->
+        <!--<%= t 'thredded.preferences.form.notifications_for_followed_topics.hint' %>-->
+      <!--</p>-->
+      <%= f.fields_for :notifications_for_followed_topics do |fn| %>
+        <%- Thredded.notifiers.each do |notifier| %>
+          <%= fn.label notifier.key do %>
+            <%= fn.check_box notifier.key, checked: f.object.notifications_for_followed_topics[notifier.key] ?
+                "checked" : false, 'data-thredded-bound-messageboard-pref' =>
+                "user_preferences_form[messageboard_notifications_for_followed_topics][#{notifier.key}]" %>
+            <%= t(notifier.i18n_key) %>
+          <% end %>
+        <%- end %>
+      <%- end %>
     </li>
     <li>
-      <%= f.label :notify_on_message do %>
-        <%= f.check_box :notify_on_message %>
-        <%= t 'thredded.preferences.form.notify_on_message.label' %>
-        <p class="thredded--form-list--hint">
-          <%= t 'thredded.preferences.form.notify_on_message.hint' %>
-        </p>
+      <label><%= t 'thredded.preferences.form.notifications_for_private_topics.label' %></label>
+      <!--<p class="thredded&#45;&#45;form-list&#45;&#45;hint">-->
+        <!--<%= t 'thredded.preferences.form.notifications_for_private_topics.hint' %>-->
+      <!--</p>-->
+      <%= f.fields_for :notifications_for_private_topics do |fn| %>
+        <%- Thredded.notifiers.each do |notifier| %>
+          <%= fn.label notifier.key do %>
+            <%= fn.check_box notifier.key, checked: f.object.notifications_for_private_topics[notifier.key] ? "checked" : false %>
+            <%= t(notifier.i18n_key)
```

Comment:
Thanks -- that clarifies things a lot. Thinking about a Thredded::FooNotifier gem -- if it is to contribute locale files does that mean it would either need to be an engine? Seems a bit over the top for a single string. Alternative I guess is that it could be just a gem provide a reasonable English default and give instructions for adding locales in main_app.